<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Generate Quality Surrogate Variable Analysis for Degradation Correction â€¢ qsvaR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="Generate Quality Surrogate Variable Analysis for Degradation Correction">
<meta property="og:description" content="The qsvaR package contains functions for removing the effect of degration in rna-seq data from postmortem brain tissue. The package is equipped to help users generate principal components associated with degradation. The components can be used in differential expression analysis to remove the effects of degradation.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">qsvaR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/Intro_qsvaR.html">Introduction to qsvaR</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LieberInstitute/qsvaR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="qsvar">qsvaR<a class="anchor" aria-label="anchor" href="#qsvar"></a>
</h1></div>
<!-- badges: start -->

<p>Differential expressions analysis requires the ability to normalize complex datasets. In the case of postmortem brain tissue we are tasked with removing the effects of bench degradation. The <code>qsvaR</code> package combines an established method for removing the effects of degradation from RNA-seq data with easy to use functions. It is the second iteration of the qSVA framework (<a href="https://doi.org/10.1073/pnas.1617384114" class="external-link">Jaffe et al, PNAS, 2017</a>).</p>
<p>The first step in the <code>qsvaR</code> workflow is to create an <a href="https://www.rdocumentation.org/packages/SummarizedExperiment/versions/1.2.3/topics/RangedSummarizedExperiment-class" class="external-link"><code>RangedSummarizedExperiment</code></a> object with the transcripts identified in our qSVA experiment. If you already have a <a href="https://www.rdocumentation.org/packages/SummarizedExperiment/versions/1.2.3/topics/RangedSummarizedExperiment-class" class="external-link"><code>RangedSummarizedExperiment</code></a> of transcripts we can do this with the <code><a href="reference/getDegTx.html">getDegTx()</a></code> function as shown below.If not this can be generated with the <a href="http://research.libd.org/SPEAQeasy/index.html" class="external-link"><code>SPEAQeasy</code></a> (a RNA-seq pipeline maintained by our lab) pipeline using the <code>--qsva</code> flag. If you already have a <a href="https://www.rdocumentation.org/packages/SummarizedExperiment/versions/1.2.3/topics/RangedSummarizedExperiment-class" class="external-link"><code>RangedSummarizedExperiment</code></a> object with transcripts then you do not need to run <a href="http://research.libd.org/SPEAQeasy/index.html" class="external-link"><code>SPEAQeasy</code></a>. This flag requires a full path to a text file, containing one Ensembl transcript ID per line for each transcript desired in the final transcripts R output object (called <code>rse_tx</code>). The <code>sig_transcripts</code> argument in this package should contain the same Ensembl transcript IDs as the text file for the <code>--qsva</code> flag.The goal of <code>qsvaR</code> is to provide software that can remove the effects of bench degradation from RNA-seq data.</p>
<div class="section level2">
<h2 id="installation-instructions">Installation Instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h2>
<p>Get the latest stable R release from CRAN. Then install <code>qsvaR</code> using from Bioconductor the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"qsvaR"</span><span class="op">)</span></span></code></pre></div>
<p>And the development version from GitHub with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"LieberInstitute/qsvaR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p>This is a basic example which shows how to obtain the quality surrogate variables (qSVs) for the brainseq <a href="http://eqtl.brainseq.org/phase2" class="external-link">phase II dataset</a>. qSVs are essentially principal components from an rna-seq experiment designed to model bench degradation. For more on principal components you can read and introductory article <a href="https://towardsdatascience.com/tidying-up-with-pca-an-introduction-to-principal-components-analysis-f876599af383#:~:text=The%20goal%20of%20PCA%20is,eliminate%20ones%20that%20do%20not" class="external-link">here</a>. At the start of this script we will have an <a href="https://www.rdocumentation.org/packages/SummarizedExperiment/versions/1.2.3/topics/RangedSummarizedExperiment-class" class="external-link"><code>RangedSummarizedExperiment</code></a> and a list of all the transcripts found in our degradation study. At the end we will have a table with differential expression results that is adjusted for qSVs.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## R packages we'll use</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/qsvaR" class="external-link">"qsvaR"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioinf.wehi.edu.au/limma/" class="external-link">"limma"</a></span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/qsvaR" class="external-link">"qsvaR"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## We'll download example data from the BrainSeq Phase II project</span></span>
<span><span class="co">## described at http://eqtl.brainseq.org/phase2/.</span></span>
<span><span class="co">##</span></span>
<span><span class="co">## We'll use BiocFileCache to cache these files so you don't have to download</span></span>
<span><span class="co">## them again for other examples.</span></span>
<span><span class="va">bfc</span> <span class="op">&lt;-</span> <span class="fu">BiocFileCache</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">BiocFileCache</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">rse_file</span> <span class="op">&lt;-</span> <span class="fu">BiocFileCache</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">bfcrpath</a></span><span class="op">(</span></span>
<span>    <span class="st">"https://s3.us-east-2.amazonaws.com/libd-brainseq2/rse_tx_unfiltered.Rdata"</span>,</span>
<span>    x <span class="op">=</span> <span class="va">bfc</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Now that we have the data in our computer, we can load it.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">rse_file</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading objects:</span></span>
<span><span class="co">#&gt;   rse_tx</span></span></code></pre></div>
<p>In this next step we subset for the transcripts associated with degradation. These were determined by Joshua M. Stolz et al, 2022. We have provided three models to choose from. Here the names <code>"cell_component"</code>, <code>"top1500"</code>, and <code>"standard"</code> refer to models that were determined to be effective in removing degradation effects. The <code>"standard"</code> model involves taking the union of the top 1000 transcripts associated with degradation from the interaction model and the main effect model. The <code>"top1500"</code> model is the same as the <code>"standard"</code> model except the union of the top 1500 genes associated with degradation is selected. The most effective of our models, <code>"cell_component"</code>, involved deconvolution of the degradation matrix to determine the proportion of cell types within our studied tissue. These proportions were then added to our <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code> and the union of the top 1000 transcripts in the interaction model, the main effect model, and the cell proportions model were used to generate this model of qSVs. In this example we will choose <code>"cell_component"</code> when using the <code><a href="reference/getDegTx.html">getDegTx()</a></code> and <code><a href="reference/select_transcripts.html">select_transcripts()</a></code> functions.</p>
<div class="figure">
<img src="./reference/figures/transcripts_venn_diagramm.png" alt="The above venn diagram shows the overlap between transcripts in each of the previously mentioned models." width="100%"><p class="caption">
The above venn diagram shows the overlap between transcripts in each of the previously mentioned models.
</p>
</div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Next we get the degraded transcripts for qSVA from the "cell_component"</span></span>
<span><span class="co">## model</span></span>
<span><span class="va">DegTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/getDegTx.html">getDegTx</a></span><span class="op">(</span><span class="va">rse_tx</span>, type <span class="op">=</span> <span class="st">"cell_component"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Now we can compute the Principal Components (PCs) of the degraded</span></span>
<span><span class="co">## transcripts</span></span>
<span><span class="va">pcTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/getPCs.html">getPCs</a></span><span class="op">(</span><span class="va">DegTx</span>, <span class="st">"tpm"</span><span class="op">)</span></span></code></pre></div>
<p>Next we use the <code><a href="reference/k_qsvs.html">k_qsvs()</a></code> function to calculate how many PCs we will need to account for the variation. A model matrix accounting for relevant variables should be used. Common variables such as Age, Sex, Ancestry are often included in the model. Again we are using our <code>RangedSummarizedExperiment</code> <code>DegTx</code> as the <code>rse_tx</code> option. Next we specify the <code>mod</code> with our <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code>. <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code> creates a design (or model) matrix, e.g., by expanding factors to a set of dummy variables (depending on the contrasts) and expanding interactions similarly. For more information on creating a design matrix for your experiment see the documentation <a href="http://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/designmatrices.html" class="external-link">here</a>. Again we use the <code>assayname</code> option to specify the we are using the <code>tpm</code> assay, where TPM stands for <em>transcripts per million</em>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Using a simple statistical model we determine the number of PCs needed (k)</span></span>
<span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Dx</span> <span class="op">+</span> <span class="va">Age</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Race</span> <span class="op">+</span> <span class="va">Region</span>,</span>
<span>    data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="va">k</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/k_qsvs.html">k_qsvs</a></span><span class="op">(</span><span class="va">DegTx</span>, <span class="va">mod</span>, <span class="st">"tpm"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">k</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 34</span></span></code></pre></div>
<p>Now that we have our PCs and the number we need we can generate our qSVs.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Obtain the k qSVs</span></span>
<span><span class="va">qsvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_qsvs.html">get_qsvs</a></span><span class="op">(</span><span class="va">pcTx</span>, <span class="va">k</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">qsvs</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 900  34</span></span></code></pre></div>
<p>This can be done in one step with our wrapper function <code>qSVA</code> which just combinds all the previous mentioned functions.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Example use of the wrapper function qSVA()</span></span>
<span><span class="va">qsvs_wrapper</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/qSVA.html">qSVA</a></span><span class="op">(</span>rse_tx <span class="op">=</span> <span class="va">rse_tx</span>, type <span class="op">=</span> <span class="st">"cell_component"</span>, mod <span class="op">=</span> <span class="va">mod</span>, assayname <span class="op">=</span> <span class="st">"tpm"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">qsvs_wrapper</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 900  34</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="differential-expression">Differential Expression<a class="anchor" aria-label="anchor" href="#differential-expression"></a>
</h2>
<p>Next we can use a standard <code>limma</code> package approach to do differential expression on the data. The key here is that we add our qSVs to the statistical model we use through <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code>. Here we input our <a href="https://www.rdocumentation.org/packages/SummarizedExperiment/versions/1.2.3/topics/RangedSummarizedExperiment-class" class="external-link"><code>Ranged SummarizedExperiment</code></a> object and our <code>model.matrix</code> with qSVs. Note here that the <code>Ranged SummarizedExperiment</code> object is the original object loaded with the full list of transcripts, not the the one we subsetted for qSVs. This is because while PCs can be generated from a subset of genes, differential expression is best done on the full dataset. The expected output is a <code>sigTx</code> object that shows the results of differential expression.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioinf.wehi.edu.au/limma/" class="external-link">"limma"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Add the qSVs to our statistical model</span></span>
<span><span class="va">mod_qSVA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span></span>
<span>    <span class="va">mod</span>,</span>
<span>    <span class="va">qsvs</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Extract the transcript expression values and put them in the</span></span>
<span><span class="co">## log2(TPM + 1) scale</span></span>
<span><span class="va">txExprs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">tpm</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Run the standard linear model for differential expression</span></span>
<span><span class="va">fitTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">txExprs</span>, <span class="va">mod_qSVA</span><span class="op">)</span></span>
<span><span class="va">eBTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fitTx</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Extract the differential expression results</span></span>
<span><span class="va">sigTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">eBTx</span>,</span>
<span>    coef <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    p.value <span class="op">=</span> <span class="fl">1</span>, number <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Explore the top results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sigTx</span><span class="op">)</span></span>
<span><span class="co">#&gt;                         logFC   AveExpr         t      P.Value    adj.P.Val</span></span>
<span><span class="co">#&gt; ENST00000553142.5 -0.06547988 2.0390889 -5.999145 2.921045e-09 0.0005786386</span></span>
<span><span class="co">#&gt; ENST00000552074.5 -0.12911383 2.4347985 -5.370828 1.009549e-07 0.0099992338</span></span>
<span><span class="co">#&gt; ENST00000510632.1  0.08994392 0.9073516  4.920042 1.037016e-06 0.0473143146</span></span>
<span><span class="co">#&gt; ENST00000530589.1 -0.10297938 2.4271711 -4.918806 1.043399e-06 0.0473143146</span></span>
<span><span class="co">#&gt; ENST00000572236.1 -0.05358333 0.6254025 -4.819980 1.697403e-06 0.0473143146</span></span>
<span><span class="co">#&gt; ENST00000450454.6  0.08446871 1.0042440  4.816539 1.726143e-06 0.0473143146</span></span>
<span><span class="co">#&gt;                           B</span></span>
<span><span class="co">#&gt; ENST00000553142.5 10.200286</span></span>
<span><span class="co">#&gt; ENST00000552074.5  6.767821</span></span>
<span><span class="co">#&gt; ENST00000510632.1  4.524039</span></span>
<span><span class="co">#&gt; ENST00000530589.1  4.518145</span></span>
<span><span class="co">#&gt; ENST00000572236.1  4.051142</span></span>
<span><span class="co">#&gt; ENST00000450454.6  4.035041</span></span></code></pre></div>
<p>Finally, you can compare the resulting t-statistics from your differential expression model against the degradation time t-statistics adjusting for the six different brain regions. This type of plot is called <code>DEqual</code> plot and was shown in the initial qSVA framework paper (<a href="https://doi.org/10.1073/pnas.1617384114" class="external-link">Jaffe et al, PNAS, 2017</a>). We are really looking for two patterns exemplified here in Figure 1 (cartoon shown earlier). A direct positive correlation with degradation shown in Figure 1 on the right tells us that there is signal in the data associated with qSVs. An example of nonconfounded data or data that has been modeled can be seen in Figure 1 on the right with its lack of relationship between the x and y variables.</p>
<div class="figure">
<img src="./reference/figures/DEqual_example.png" alt="Cartoon showing patterns in DEqual plots" width="100%"><p class="caption">
Cartoon showing patterns in DEqual plots
</p>
</div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Generate a DEqual() plot using the model results with qSVs</span></span>
<span><span class="fu"><a href="reference/DEqual.html">DEqual</a></span><span class="op">(</span><span class="va">sigTx</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="reference/figures/README-DEqual-1.png" alt="Result of Differential Expression with qSVA normalization." width="100%"><p class="caption">
Result of Differential Expression with qSVA normalization.
</p>
</div>
<p>For comparison, here is the <code><a href="reference/DEqual.html">DEqual()</a></code> plot for the model without qSVs.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Generate a DEqual() plot using the model results without qSVs</span></span>
<span><span class="fu"><a href="reference/DEqual.html">DEqual</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">txExprs</span>, <span class="va">mod</span><span class="op">)</span><span class="op">)</span>, coef <span class="op">=</span> <span class="fl">2</span>, p.value <span class="op">=</span> <span class="fl">1</span>, number <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="reference/figures/README-DEqual-no-qSVs-1.png" alt="Result of Differential Expression without qSVA normalization." width="100%"><p class="caption">
Result of Differential Expression without qSVA normalization.
</p>
</div>
<p>In these two DEqual plots we can see that the first is much better. With a correlation of -0.014 we can effectively conclude that we have removed the effects of degradation from the data. In the second plot after modeling for several common variables we still have a correlation of 0.5 with the degradation experiment. This high correlation shows we still have a large amount of signal from degradation in our data potentially confounding our case-control (SCZD vs neurotypical controls) differential expression results.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/qsvaR" class="external-link">View on Bioconductor</a></li>
<li><a href="https://github.com/LieberInstitute/qsvaR/" class="external-link">Browse source code</a></li>
<li><a href="https://support.bioconductor.org/t/qsvaR" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0" class="external-link">Artistic-2.0</a></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
<li><a href="SUPPORT.html">Getting help</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing qsvaR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Joshua Stolz <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0001-5694-5247" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Hedia Tnani <br><small class="roles"> Contributor, maintainer </small> <a href="https://orcid.org/0000-0002-0380-9740" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#stable" class="external-link"><img src="https://img.shields.io/badge/lifecycle-stable-brightgreen.svg" alt="Lifecycle: stable"></a></li>
<li><a href="https://bioconductor.org/checkResults/release/bioc-LATEST/qsvaR" class="external-link"><img src="http://www.bioconductor.org/shields/build/release/bioc/qsvaR.svg" alt="Bioc release status"></a></li>
<li><a href="https://bioconductor.org/checkResults/devel/bioc-LATEST/qsvaR" class="external-link"><img src="http://www.bioconductor.org/shields/build/devel/bioc/qsvaR.svg" alt="Bioc devel status"></a></li>
<li><a href="http://bioconductor.org/packages/stats/bioc/qsvaR/" class="external-link"><img src="https://bioconductor.org/shields/downloads/release/qsvaR.svg" alt="Bioc downloads rank"></a></li>
<li><a href="https://support.bioconductor.org/tag/qsvaR" class="external-link"><img src="https://bioconductor.org/shields/posts/qsvaR.svg" alt="Bioc support"></a></li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/qsvaR.html#since" class="external-link"><img src="https://bioconductor.org/shields/years-in-bioc/qsvaR.svg" alt="Bioc history"></a></li>
<li><a href="http://bioconductor.org/checkResults/devel/bioc-LATEST/qsvaR/" class="external-link"><img src="https://bioconductor.org/shields/lastcommit/devel/bioc/qsvaR.svg" alt="Bioc last commit"></a></li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/qsvaR.html#since" class="external-link"><img src="https://bioconductor.org/shields/dependencies/release/qsvaR.svg" alt="Bioc dependencies"></a></li>
<li><a href="https://codecov.io/gh/LieberInstitute/qsvaR?branch=devel" class="external-link"><img src="https://codecov.io/gh/LieberInstitute/qsvaR/branch/devel/graph/badge.svg" alt="Codecov test coverage"></a></li>
<li><a href="https://github.com/LieberInstitute/qsvaR/actions" class="external-link"><img src="https://github.com/LieberInstitute/qsvaR/workflows/R-CMD-check-bioc/badge.svg" alt="R build status"></a></li>
<li><a href="https://github.com/LieberInstitute/qsvaR/issues" class="external-link"><img src="https://img.shields.io/github/issues/LieberInstitute/qsvaR" alt="GitHub issues"></a></li>
<li><a href="https://github.com/LieberInstitute/qsvaR/pulls" class="external-link"><img src="https://img.shields.io/github/issues-pr/LieberInstitute/qsvaR" alt="GitHub pulls"></a></li>
<li><a href="https://zenodo.org/badge/latestdoi/421556636" class="external-link"><img src="https://zenodo.org/badge/421556636.svg" alt="DOI"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Joshua Stolz, Hedia Tnani.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
