<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Generate Quality Surrogate Variable Analysis for Degradation Correction â€¢ qsvaR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Generate Quality Surrogate Variable Analysis for Degradation Correction">
<meta property="og:description" content="The qsvaR package contains functions for removing the effect of degration in rna-seq data from postmortem brain tissue. The package is equipped to help users generate principal components associated with degradation. The components can be used in differential expression analysis to remove the effects of degradation.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">qsvaR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/Intro_qsvaR.html">Introduction to qsvaR</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LieberInstitute/qsvaR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="qsvar">qsvaR<a class="anchor" aria-label="anchor" href="#qsvar"></a>
</h1></div>
<!-- badges: start -->
<!-- badges: end -->
<p>Differential expressions analysis requires the ability normalize
complex datasets. In the case of postmortem brain tissue we are tasked
with removing the effects of bench degradation. The qsvaR package
combines an established method for removing the effects of degradation
from RNA-seq data with easy to use functions. The first step in this
workflow is to create an <code>Ranged Summarized Experiment</code>[<a href="https://www.rdocumentation.org/packages/SummarizedExperiment/versions/1.2.3/topics/RangedSummarizedExperiment-class" class="external-link uri">https://www.rdocumentation.org/packages/SummarizedExperiment/versions/1.2.3/topics/RangedSummarizedExperiment-class</a>]
object with the transcripts identified in our qsva experiment. If you
already have a <code>Ranged Summarized Experiment</code>[<a href="https://www.rdocumentation.org/packages/SummarizedExperiment/versions/1.2.3/topics/RangedSummarizedExperiment-class" class="external-link uri">https://www.rdocumentation.org/packages/SummarizedExperiment/versions/1.2.3/topics/RangedSummarizedExperiment-class</a>]
of transcripts we can do this with the getDegTx function as shown
below.If not this can be generated with the <code>SPEAQeasy</code> (a
RNA-seq pipeline maintained by our lab) pipeline usinge the
<code>--qsva</code> flag. If you already have a
<code>Ranged Summarized Experiment</code>[<a href="https://www.rdocumentation.org/packages/SummarizedExperiment/versions/1.2.3/topics/RangedSummarizedExperiment-class" class="external-link uri">https://www.rdocumentation.org/packages/SummarizedExperiment/versions/1.2.3/topics/RangedSummarizedExperiment-class</a>]
object with transcripts then you do not need to run
<code>SPEAQeasy</code>. This flag requires a full path to a text file,
containing one Ensembl transcript ID per line for each transcript
desired in the final transcripts R output object (called
<code>rse_tx</code>). The <code>sig_transcripts</code> argument in this
package should contain the same Ensembl transcript IDs as the text file
for the <code>--qsva</code> flag.The goal of <code>qsvaR</code> is to
provide software that can remove the effects of bench degradation from
RNA-seq data.</p>
<div class="section level2">
<h2 id="installation-instructions">Installation Instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h2>
<p>Get the latest stable R release from CRAN. Then install
<code>qsvaR</code> using from Bioconductor the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="op">}</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"qsvaR"</span><span class="op">)</span></code></pre></div>
<p>And the development version from GitHub with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("devtools")</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"LieberInstitute/qsvaR"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p>This is a basic example which shows you how to obtain the quality
surrogate variables (qSVs) for the brainseq
<code>phase II dataset</code>[<a href="http://eqtl.brainseq.org/phase2" class="external-link uri">http://eqtl.brainseq.org/phase2</a>]. qSVs are essentially
pricipal components from an rna-seq experiment designed to model bench
degradation. For more on principal components you can read and
introductory article <code>here</code>[<a href="https://towardsdatascience.com/tidying-up-with-pca-an-introduction-to-principal-components-analysis-f876599af383" class="external-link uri">https://towardsdatascience.com/tidying-up-with-pca-an-introduction-to-principal-components-analysis-f876599af383</a>].
At the start of this script we will have a
<code>Ranged SummmarizedExperiment</code>[<a href="https://www.rdocumentation.org/packages/SummarizedExperiment/versions/1.2.3/topics/RangedSummarizedExperiment-class" class="external-link uri">https://www.rdocumentation.org/packages/SummarizedExperiment/versions/1.2.3/topics/RangedSummarizedExperiment-class</a>]
and a list of all the transcripts found in our degradation study. At the
end we will have a table with differential expression results that is
adjusted for qSVs.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## R packages we'll use</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/qsvaR" class="external-link">"qsvaR"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="http://bioinf.wehi.edu.au/limma" class="external-link">"limma"</a></span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/LieberInstitute/qsvaR" class="external-link">"qsvaR"</a></span><span class="op">)</span>

<span class="co">## We'll download example data from the BrainSeq Phase II project</span>
<span class="co">## described at http://eqtl.brainseq.org/phase2/.</span>
<span class="co">##</span>
<span class="co">## We'll use BiocFileCache to cache these files so you don't have to download</span>
<span class="co">## them again for other examples.</span>
<span class="va">bfc</span> <span class="op">&lt;-</span> <span class="fu">BiocFileCache</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">BiocFileCache</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">rse_file</span> <span class="op">&lt;-</span> <span class="fu">BiocFileCache</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocFileCache/man/BiocFileCache-class.html" class="external-link">bfcrpath</a></span><span class="op">(</span><span class="st">"https://s3.us-east-2.amazonaws.com/libd-brainseq2/rse_tx_unfiltered.Rdata"</span>, x <span class="op">=</span> <span class="va">bfc</span><span class="op">)</span>

<span class="co">## Now that we have the data in our computer, we can load it.</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">rse_file</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; Loading objects:</span>
<span class="co">#&gt;   rse_tx</span>

<span class="co">## Next we get the degraded transcripts for qSVA</span>
<span class="va">DegTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/getDegTx.html">getDegTx</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span>

<span class="co">## Now we can compute the Principal Components (PCs) of the degraded transcripts</span>
<span class="va">pcTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/getPCs.html">getPCs</a></span><span class="op">(</span><span class="va">DegTx</span>, <span class="st">"tpm"</span><span class="op">)</span>

<span class="co">## Using a simple statistical model we determine the number of PCs needed (k)</span>
<span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Dx</span> <span class="op">+</span> <span class="va">Age</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Race</span> <span class="op">+</span> <span class="va">Region</span>,
    data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span>
<span class="op">)</span>
<span class="va">k</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/k_qsvs.html">k_qsvs</a></span><span class="op">(</span><span class="va">DegTx</span>, <span class="va">mod</span>, <span class="st">"tpm"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">k</span><span class="op">)</span>
<span class="co">#&gt; [1] 34</span></code></pre></div>
<p>Now that we have our PCs and the number we need we can generate our
qSVs.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Obtain the k qSVs</span>
<span class="va">qsvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_qsvs.html">get_qsvs</a></span><span class="op">(</span><span class="va">pcTx</span>, <span class="va">k</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">qsvs</span><span class="op">)</span>
<span class="co">#&gt; [1] 900  34</span></code></pre></div>
<p>This can be done in one step with our wrapper function
<code>qSVA</code> which just combinds all the previous mentioned
functions.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">qsvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/qSVA.html">qSVA</a></span><span class="op">(</span>rse_tx <span class="op">=</span> <span class="va">rse_tx</span>, type <span class="op">=</span> <span class="st">"cell_component"</span>, mod <span class="op">=</span> <span class="va">mod</span>, assayname <span class="op">=</span> <span class="st">"tpm"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">qsvs</span><span class="op">)</span>
<span class="co">#&gt; [1] 900  34</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="differential-expression">Differential Expression<a class="anchor" aria-label="anchor" href="#differential-expression"></a>
</h2>
<p>Next we can use a standard <code>limma</code> package approach to do
differential expression on the data. The key here is that we add our
qSVs to the statisical model we use through <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code>.
Here we input our <code>Ranged SummarizedExperiment</code> object and
our <code>model.matrix</code> with qSVs. The expected output is a sigTx
object that shows the results of differential expression.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="http://bioinf.wehi.edu.au/limma" class="external-link">"limma"</a></span><span class="op">)</span>

<span class="co">## Add the qSVs to our statistical model</span>
<span class="va">mod_qSVA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span>
    <span class="va">mod</span>,
    <span class="va">qsvs</span>
<span class="op">)</span>

<span class="co">## Extract the transcript expression values and put them in the</span>
<span class="co">## log2(TPM + 1) scale</span>
<span class="va">txExprs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">tpm</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span>

<span class="co">## Run the standard linear model for differential expression</span>
<span class="va">fitTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">txExprs</span>, <span class="va">mod_qSVA</span><span class="op">)</span>
<span class="va">eBTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fitTx</span><span class="op">)</span>
<span class="co">#&gt; Warning: Zero sample variances detected, have been offset away from zero</span>

<span class="co">## Extract the differential expression results</span>
<span class="va">sigTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">eBTx</span>,
    coef <span class="op">=</span> <span class="fl">2</span>,
    p.value <span class="op">=</span> <span class="fl">1</span>, number <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sigTx</span><span class="op">)</span>
<span class="co">#&gt;                         logFC   AveExpr         t      P.Value    adj.P.Val</span>
<span class="co">#&gt; ENST00000553142.5 -0.06547988 2.0390889 -5.999145 2.921045e-09 0.0005786386</span>
<span class="co">#&gt; ENST00000552074.5 -0.12911383 2.4347985 -5.370828 1.009549e-07 0.0099992338</span>
<span class="co">#&gt; ENST00000510632.1  0.08994392 0.9073516  4.920042 1.037016e-06 0.0473143146</span>
<span class="co">#&gt; ENST00000530589.1 -0.10297938 2.4271711 -4.918806 1.043399e-06 0.0473143146</span>
<span class="co">#&gt; ENST00000572236.1 -0.05358333 0.6254025 -4.819980 1.697403e-06 0.0473143146</span>
<span class="co">#&gt; ENST00000450454.6  0.08446871 1.0042440  4.816539 1.726143e-06 0.0473143146</span>
<span class="co">#&gt;                           B</span>
<span class="co">#&gt; ENST00000553142.5 10.200286</span>
<span class="co">#&gt; ENST00000552074.5  6.767821</span>
<span class="co">#&gt; ENST00000510632.1  4.524039</span>
<span class="co">#&gt; ENST00000530589.1  4.518145</span>
<span class="co">#&gt; ENST00000572236.1  4.051142</span>
<span class="co">#&gt; ENST00000450454.6  4.035041</span></code></pre></div>
<p>Finally, you can compare the resulting t-statistics from your
differential expression model against the degradation time t-statistics
adjusting for the six different brain regions. This type of plot is
called <code>DEqual</code> plot and was shown in the initial qSVA
framework paper (<a href="https://doi.org/10.1073/pnas.1617384114" class="external-link">Jaffe
et al, PNAS, 2017</a>). We are really looking for two patterns
exemplified here in figure 1. A direct positive correlation with
degradation shown in figure 1 on the right tells us that there is signal
in the data associated with qSVs. An example of nonconfounded data or
data that has been modeled can be seen in figure 1 on the right with its
lack of relationship between the x and y variables.</p>
<div class="figure">
<img src="./reference/figures/DEqual_example.png" alt=""><p class="caption">figure</p>
</div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/DEqual.html">DEqual</a></span><span class="op">(</span><span class="va">sigTx</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-DEqual-1.png" width="100%"></p>
<p>For comparison, here is the <code><a href="reference/DEqual.html">DEqual()</a></code> plot for the model
without qSVs.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/DEqual.html">DEqual</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">txExprs</span>, <span class="va">mod</span><span class="op">)</span><span class="op">)</span>, coef <span class="op">=</span> <span class="fl">2</span>, p.value <span class="op">=</span> <span class="fl">1</span>, number <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Warning: Zero sample variances detected, have been offset away from zero</span></code></pre></div>
<p><img src="reference/figures/README-DEqual-no-qSVs-1.png" width="100%"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/LieberInstitute/qsvaR/" class="external-link">Browse source code</a></li>
<li><a href="https://support.bioconductor.org/t/qsvaR" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0" class="external-link">Artistic-2.0</a></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing qsvaR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Joshua Stolz <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-5694-5247" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Joshua Stolz.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
