---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    fig.path = "man/figures/README-",
    out.width = "100%"
)
```

# qsvaR

<!-- badges: start -->
<!-- badges: end -->

The goal of qsvaR is to provide software that can remove the effects of bench degradation from RNA-seq data.

## Installation Instructions
Get the latest stable R release from CRAN. Then install DeconvoBuddies using from Bioconductor the following code:
```r
if (!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
}

BiocManager::install("DeconvoBuddies")
```
And the development version from GitHub with:

```r
# install.packages("devtools")
devtools::install_github("LieberInstitute/qsvaR")
```


## Example

This is a basic example which shows you how to obtain the qsvs for a dataset:

```{r example_qsvs}
library(qsvaR)
## basic example data
bfc <- BiocFileCache::BiocFileCache()
rse_file <- BiocFileCache::bfcrpath("https://s3.us-east-2.amazonaws.com/libd-brainseq2/rse_tx_unfiltered.Rdata", x = bfc)
load(rse_file,verbose = TRUE)

##get degraded transcripts for qsva
DegTx<-getDegTx(rse_tx,rownames(covComb_tx_deg))

#get pcs of degraded
pcTx<-getPCs(DegTx, "tpm")

#use a simple model and our get K function to determine the number of pcs needed
mod <-model.matrix(~ Dx + Age +Sex + Race + Region,
    data = colData(rse_tx))
k<-k_qsvs(DegTx,mod, "tpm")
print(k)
```
Now that we have our pcs and the number we need we can generate our qsvs

```{r exampleDE}
qsvs<-get_qsvs(pcTx, k)
dim(qsvs)
```

What is special about using `README.Rmd` instead of just `README.md`? You can include R chunks like so:

###Differential Expression
Next we can use a standard limma package approach to do differential expression on the data. The key here is that we add our qsvs to the model.matrix.

```{r "perform DE"}
    library(limma)
    pd<-colData(rse_tx)
    pd<-cbind(qsvs,pd)
    mod <-model.matrix(~ Dx + Age + Sex + Race + Region + pd$PC1 +pd$PC2+ pd$PC3 +pd$PC4+ pd$PC5 +pd$PC6+ pd$PC7 +pd$PC8+ pd$PC9 +pd$PC10+ pd$PC11 +pd$PC12+ pd$PC13 +pd$PC14+ pd$PC15 +pd$PC16 + pd$PC17 +pd$PC18+ pd$PC19 +pd$PC20+ pd$PC21 +pd$PC22+ pd$PC23 +pd$PC24+ pd$PC25 +pd$PC26+ pd$PC27 +pd$PC28+ pd$PC29 +pd$PC30+ pd$PC31, data = colData(rse_tx))
    txExprs <- log2(assays(rse_tx)$tpm + 1)
    fitTx <- lmFit(txExprs, mod)
    eBTx <- eBayes(fitTx)
    sigTx <- topTable(eBTx,
    coef = 2,
    p.value = 1, number = nrow(rse_tx)
)
    head(sigTx)
```


