# Quick start to using to `qsvaR`

```{r "start", message=FALSE}
library("qsvaR")
library(jaffelab)
library(limma)
```
load("../rse_tx_unfiltered.Rdata")
Edit this as you see fit =)

Here is an example of you can cite your package inside the vignette:

* `r Biocpkg("qsvaR")` `r Citep(bib[["qsvaR"]])`

##Get QSVs

###Significant Transcripts

Differential expressions analysis requires the ability normalize complex datasets. In the case of postmortem brain tissue we are tasked with removing the effects of bench degradation. The qsvaR package combines an established method for removing the effects of degradation from RNA-seq data with easy to use functions. The first step in this workflow is to create an Ranged Summarized Experiment object with the transcripts identified in our qsva experiment. We can do this with the getDegTx function as shown below. sig_transcripts is a list of transcripts identified to be associated with degradation. In this example we use the lieber institutes brainseq phase 2 data downloaded from https://s3.us-east-2.amazonaws.com/libd-brainseq2/rse_tx_unfiltered.Rdata

```{r `getDegTx demo`}
load("../../rse_tx_unfiltered.Rdata")
DegTx<-getDegTx(rse_tx,rownames(covComb_tx_deg))
dim(DegTx)
```

###Get principal components
The qsvs are derived from taking the principal components of the selected transcript expression data. This can be done with the function getBonfTx. Here "tpm" is the name of the assay we are using.

```{r `getBonfTx demo`}
pcTx<-getBonfTx(DegTx, "tpm")
```
Next we use the k_qsvs function to calculate how many pcs we will need to acount for the variation. A model matrix accounting for relevant variables should be used. Common options are listed below.

```{r `k_qsvs demo`}
mod_tx <-model.matrix(~ Dx + Age + mitoRate + rRNA_rate + totalAssignedGene,
    data = colData(rse_tx))
k<-k_qsvs(DegTx,mod_tx, "tpm")
print(k)
```

Finally we subest our data to the calculated number of pcs.
```{r `get_qsvs demo`}
qsvs<-get_qsvs(pcTx, k)
dim(qsvs)
```

```{r `perform DE`}
    pd<-colData(rse_tx)
    pd<-cbind(qsvs,pd)
    mod <-model.matrix(~ Dx + Age + mitoRate + rRNA_rate + totalAssignedGene + pd$PC1 +pd$PC2+ pd$PC3 +pd$PC4+ pd$PC5 +pd$PC6+ pd$PC7 +pd$PC8+ pd$PC9 +pd$PC10+ pd$PC11 +pd$PC12+ pd$PC13 +pd$PC14+ pd$PC15 +pd$PC16 + pd$PC17 +pd$PC18+ pd$PC19 +pd$PC20+ pd$PC21 +pd$PC22+ pd$PC23 +pd$PC24+ pd$PC25 +pd$PC26+ pd$PC27 +pd$PC28+ pd$PC29 +pd$PC30+ pd$PC31, data = colData(rse_tx))
   txExprs = log2(assays(rse_tx)$tpm+1)
fitTx = lmFit(txExprs,mod)
eBTx = eBayes(fitTx)
sigTx = topTable(eBTx,coef=2,
	p.value = 1,number=nrow(rse_tx))
head(sigTx)
```
title: "qsvaRexample"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{qsvaRexample}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(qsvaR)
```
